utilities/user_utils.py

```diff
@@ -0,0 +1,263 @@
+
+    if login_with_user_password(api_address=cluster_url, user=username, password=password):
+        # Login was successful, get the current context
+        _, stdout, _ = run_command(command=["oc", "config", "current-context"], check=True)
+        user_context = stdout.strip()
+        LOGGER.info(f"Successfully created context for user {username}: {user_context}")
+        return user_context
+
+    LOGGER.error(f"Login failed for user {username}")
+    return None
+
+
+def _create_htpasswd_secret(username: str, password: str, secret_name: str) -> None:
+    """
+    Creates an htpasswd file and corresponding OpenShift secret for a user.
+
+    This function generates an htpasswd file for the given username and password,
+    then creates a Kubernetes secret in the 'openshift-config' namespace using that file.
+
+    Args:
+        username: The username to add to the htpasswd file.
+        password: The password for the user.
+        secret_name: The name of the secret to create.
+    """
+    with tempfile.NamedTemporaryFile(mode="w+") as temp_file:
+        subprocess.run(args=["htpasswd", "-c", "-b", temp_file.name, username, password], check=True)
+        LOGGER.info(f"Creating secret {secret_name} in openshift-config namespace")
+        run_command(
+            command=[
+                "oc",
+                "create",
+                "secret",
+                "generic",
+                secret_name,
+                f"--from-file=htpasswd={temp_file.name}",
+                "-n",
+                "openshift-config",
+            ],
+            check=True,
+        )
+        LOGGER.info(f"Created secret: {secret_name}")
+
+
+def _update_oauth_config(idp_name: str, secret_name: str) -> None:
+    """
+    Updates the OpenShift OAuth configuration to add a new HTPasswd identity provider.
+
+    Args:
+        idp_name: The name of the identity provider to add.
+        secret_name: The name of the secret containing the htpasswd file.
+    """
+    _, stdout, _ = run_command(command=["oc", "get", "oauth", "cluster", "-o", "json"], check=True)
+    oauth_config = json.loads(stdout)
+    oauth_config["spec"]["identityProviders"].append({
+        "name": idp_name,
+        "mappingMethod": "claim",
+        "type": "HTPasswd",
+        "htpasswd": {"fileData": {"name": secret_name}},
+    })
+    with tempfile.NamedTemporaryFile(mode="w", suffix=".json", delete=False) as temp_file:
+        json.dump(oauth_config, temp_file)
+        temp_file.flush()
+        run_command(command=["oc", "replace", "-f", temp_file.name], check=True)
+        os.unlink(temp_file.name)
+
+
+def _remove_idp_from_oauth(idp_name: str) -> None:
+    """
+    Removes a specified identity provider from the OpenShift OAuth configuration.
+
+    Args:
+        idp_name: The name of the identity provider to remove.
+    """
+    _, stdout, _ = run_command(command=["oc", "get", "oauth", "cluster", "-o", "json"], check=True)
```

Comment:
Ditto. please use https://github.com/RedHatQE/openshift-python-wrapper/blob/main/ocp_resources/oauth.py

Reply:
I tried and it is not really working fine with the update